File: algorithms/JavaScript/strings/palindrome.js
// JavaScript Palindrome Checker: Checks whether a word is the same in reverse. Ignores punctuation, capitalization & spaces.

function isPalindrome(str) {
    // First convert the string into proper alphanumeric word
    let properStr = str
        .replace(/[_\W]/g, "")
        .toLowerCase();
    
    // Now reverse the proper string
    let reverseStr = properStr
        .split("")
        .reverse()
        .join("");
    
    // Finally compare the proper string and reverse string and return true or false
    return properStr === reverseStr;
}

// Output to the console
console.log(isPalindrome("eye"));
console.log(isPalindrome("Mr. Owl ate my metal worm"));
console.log(isPalindrome("RAce C*_aR"));

File: algorithms/CSharp/src/Strings/palindrome.cs
using System;
using System.Linq;
using System.Text.RegularExpressions;

namespace Algorithms.Strings
{
    public class Palindrome
    {
        public static void Main()
        {
            bool result = false;
            result = IsPalindrome("abba");
            Console.WriteLine(result);
            result = IsPalindrome("abbccbbA");
            Console.WriteLine(result);
            result = IsPalindrome("Mr. Owl ate my metal worm");
            Console.WriteLine(result);
        }

        public static bool IsPalindrome(string source)
        {
            source = source.ToLower();
            source = Regex.Replace(source, @"[^0-9A-Za-z]", "");
            string reverse = new string(Enumerable.Range(1, source.Length).Select(i => source[source.Length - i]).ToArray());
            return reverse == source;
        }
    } 
}
